modules/viqeoBidAdapter.js

```diff
@@ -0,0 +1,168 @@
+import {registerBidder} from '../src/adapters/bidderFactory.js';
+import {logError, logInfo, _each, mergeDeep, isFn, isNumber, isPlainObject} from '../src/utils.js'
+import {BANNER} from '../src/mediaTypes.js';
+import {Renderer} from '../src/Renderer.js';
+
+const BIDDER_CODE = 'viqeo';
+const VIQEO_ENDPOINT = 'https://ads.betweendigital.com/openrtb_bid';
+const RENDERER_URL = 'https://cdn.viqeo.tv/js/vq_starter.js';
+const DEFAULT_CURRENCY = 'USD';
+const DEFAULT_SSPID = 44697;
+
+function getBidFloor(bid) {
+  const {floor, currency} = bid.params;
+  const curr = currency || DEFAULT_CURRENCY;
+  if (!isFn(bid.getFloor)) {
+    return {floor: isNumber(floor) ? floor : 0, currency: curr};
+  }
+  const floorInfo = bid.getFloor({currency: curr, mediaType: BANNER, size: '*'});
+  if (isPlainObject(floorInfo) && isNumber(floorInfo.floor) && floorInfo.currency === curr) {
+    return floorInfo;
+  }
+  return {floor: floor || 0, currency: currency || DEFAULT_CURRENCY};
+}
+
+/**
+ * @type {BidderSpec}
+ */
+export const spec = {
+  code: BIDDER_CODE,
+  supportedMediaTypes: [BANNER],
+  /**
+   * @param {BidRequest} bidRequest The bid params to validate.
+   * @return boolean True if this is a valid bid, and false otherwise.
+   */
+  isBidRequestValid: ({params}) => {
+    if (!params) {
+      logError('failed validation: params not declared');
+      return false;
+    }
+    if (!params.user && !params.user?.buyeruid) {
+      logError('failed validation: user.buyeruid not declared');
+      return false;
+    }
+    if (!params.playerOptions) {
+      logError('failed validation: playerOptions not declared');
+      return false;
+    }
+    const {profileId, videoId, playerId} = params.playerOptions;
+    if (!profileId) {
+      logError('failed validation: profileId not declared');
+      return false;
+    }
+    if (!videoId && !playerId) {
+      logError('failed validation: videoId or playerId not declared');
+      return false;
+    }
+    return true;
+  },
+  /**
+   * @param validBidRequests {BidRequest[]}
+   * @returns {ServerRequest[]}
+   */
+  buildRequests: (validBidRequests) => {
+    logInfo('validBidRequests', validBidRequests);
+    const bidRequests = [];
+    _each(validBidRequests, (bid, i) => {
+      const {
+        params: {test, sspId},
+        mediaTypes: {banner},
```

Comment:
This should be updated to Outstream

Reply:
@ncolletti We mix all types of demand and place it to the banner slot. Only on our server side for test purposes now is set up VAST response, but the module itself is mostly created to bring outstream and instream demand to classical banner slots. In case of instream bids win the module creates a video player with content inside. May we be approved in this case? And should we set up server side to show different types of banners?

Reply:
@ncolletti can we keep the work of adapter this way?

Reply:
fix it

Reply:
@ncolletti please check new settings